openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -477,7 +477,7 @@ def is_enabled_without_course_context(self):
         """
         return self._is_enabled()
 
-    def is_enabled(self, course_key=None):
+    def is_enabled(self, course_key):
```

Comment:
`course_key` shouldn't look like an optional parameter if we're going to raise when it is missing.

Reply:
I assume it was like that because they didn't want to so radically change the parent's signature? But as mentioned above, I'm a vote in favor of making that arg more optional. But I don't really care.

Reply:
Sanity check - is there anywhere in the codebase we were calling this without a course_key?  i.e., other, existing CourseWaffleFlags that've been set up outside of our own use cases?

Reply:
(Can't remember python's behavior if it were called as `is_enabled()` with this new method signature)

Reply:
(I've reverted this specific change.)